networkit/cpp/centrality/PageRank.cpp

```diff
@@ -5,73 +5,83 @@
  *      Author: Henning
  */
 
-#include <networkit/centrality/PageRank.hpp>
+// networkit-format
+
 #include <networkit/auxiliary/NumericTools.hpp>
 #include <networkit/auxiliary/SignalHandling.hpp>
+#include <networkit/centrality/PageRank.hpp>
 
 namespace NetworKit {
 
-PageRank::PageRank(const Graph& G, double damp, double tol):
-        Centrality(G, true), damp(damp), tol(tol)
-{
-
-}
+PageRank::PageRank(const Graph &G, double damp, double tol)
+    : Centrality(G, true), damp(damp), tol(tol) {}
 
 void PageRank::run() {
     Aux::SignalHandler handler;
-    count n = G.numberOfNodes();
-    count z = G.upperNodeIdBound();
-    double oneOverN = 1.0 / (double) n;
-    double teleportProb = (1.0 - damp) / (double) n;
-    scoreData.resize(z, oneOverN);
+    const auto n = G.numberOfNodes();
+    const auto z = G.upperNodeIdBound();
+
+    const auto teleportProb = (1.0 - damp) / static_cast<double>(n);
+    scoreData.resize(z, 1.0 / static_cast<double>(n));
     std::vector<double> pr = scoreData;
-    bool isConverged = false;
 
     std::vector<double> deg(z, 0.0);
-    G.parallelForNodes([&](node u) {
-        deg[u] = (double) G.weightedDegree(u);
+    G.parallelForNodes([&](const node u) { deg[u] = static_cast<double>(G.weightedDegree(u)); });
+
+    iterations = 0;
+
+    auto sumL1Norm = [&](const node u) { return std::abs(scoreData[u] - pr[u]); };
+
+    auto sumL2Norm = [&](const node u) {
+        const auto d = scoreData[u] - pr[u];
+        return d * d;
+    };
+
+    auto converged([&]() {
+        if (iterations >= maxIterations) {
+            return true;
+        }
+
+        if (norm == Norm::L2Norm) {
+            return std::sqrt(G.parallelSumForNodes(sumL2Norm)) <= tol;
+        }
+
+        return G.parallelSumForNodes(sumL1Norm) <= tol;
     });
 
-    while (! isConverged) {
+    bool isConverged = false;
+    while (!isConverged) {
         handler.assureRunning();
-        G.balancedParallelForNodes([&](node u) {
+        G.balancedParallelForNodes([&](const node u) {
```

Comment:
Since `auto converged ...` is moved outside the loop, maybe it is a good idea also to move the scoring-update function outside (consistency of style)?

Reply:
Maybe, the problem is that we are doing that in two steps, and I this would result in more lines of code without improving the style that much.